### ISSUE TITLE
QuickLook doesn't work with `docx` files

### ISSUE THREAD

**Issue by atanasj** [mronkko/ZoteroQuickLook#14]
QuickLook xpi doesn't work with `docx` files. Works fine with `pdf` and `pptx`. 

```
ZoteroQuickLook 1.3.2
Zotero 5.0.38
macOS 10.13.3
Word 16.10
```

Please let me know if you need any more info. Thanks for a great plugin!

**Comment by atanasj** [comment]
Yes I am. Is there anything else you suggest I do, or other information you may need to help debug this issue?

**Comment by bwiernik** [comment]
First update to the most recent versions of Zotero, ZoteroQuickLook, and Word and confirm that the error is still present.

**Comment by atanasj** [comment]
So, after updating to the following versions:

```
ZoteroQuickLook 1.3.3
Zotero 5.0.45
Word 16.2
```
The verdict is: 

I do get popup image. From a performance perspective, loading a `pdf` via `ZoteroQuickLook` takes <1 second, whereas loading a `docx` (see first screenshot below) takes ~20 seconds. I did not ever wait before for this long, so I may have got the same type of popup before. However, `ZoteroQuickLook` does not work as I would expect with `docx`. On the `pdf` files, `ZoteroQuickLook` can scroll through the document (see second screenshot below), but for `docx`, only a thumbnail of the first page is produced. When viewing a `docx` via the Mac Finder, the native quicklook functionality provides a way to scroll through the document.

Please let me if you need more info.  Thanks for your help. 


![image](https://user-images.githubusercontent.com/27214625/39182613-c869dd4a-4800-11e8-9e5f-ebb331a1e188.png)

![image](https://user-images.githubusercontent.com/27214625/39182815-70b03e72-4801-11e8-9a8b-549800177731.png)

**Comment by bwiernik** [comment]
Okay, I can replicate the same behavior on a Mac. It seems to be a consequence of using the qlmanage utility to access QuickLook from Zotero. qlmanage is a utility for testing quicklook functions, and it looks like it doesn't work correctly with .docx files (or any other qlgenerators for non-standard file types). I tried some debugging with qlmanage to try to get it to work, but couldn't. The issue isn't present when I use ZQL on Windows with the QuickLook program. So this is an issue with the utility used to show quicklook previews on Mac. I don't know whether this behavior was always present or whether it is new in High Sierra. I am not sure whether it's fixable without writing a full-blown QuickLook implementation into Zotero, which likely won't happen until Zotero migrates to the Electron platform, which has native QuickLook support.

**Comment by bwiernik** [comment]
It might be possible to at least get ZQL to skip over non-PDF files to avoid the delay in the unhelpful quicklook window popping up.

**Comment by atanasj** [comment]
Thanks for your response. 

Sounds like it's best / easiest to just be aware of the issue, and then wait for `Zotero` to migrate to `Electron`.  

It's not that big a deal really, as I don't have that many `docx` files to manage. I'm happy to live with it. 

Not sure what your process is about closing/managing this issue... I'll leave it up to you. 

Thanks once again for an awesome package!
